src/main.rs

```diff
@@ -205,25 +205,287 @@ fn main() {
                     4 => 40,
                     _ => 90
                 };
-
+				
+// This is where the magic happens!
+                
                 level.num_miis = rng.sample(Uniform::new_inclusive(4, max_miis));
+				
+				level.unk12 = 0.0;
+				level.unk13 = 0.0;
+				level.unk14 = 0.0;
+				level.unk15 = 0.0;
+				level.unk16 = 0.0;
                 level.behavior = match level.level_type {
-                    9  | 10 | 11 => if rng.gen_ratio(1, 2) { 1 } else { 4 }
-                    17 | 18 | 19 => 0,
+                    9  | 10 | 5 | 11 | 20 => if rng.gen_ratio(1, 2) { 1 } else { 3 } //Find The Odd Mii(s) out
+                    17 | 18 | 19 => 0, // Find The Insomniac 
+					14 | 15 | 16 => 0, // Find The Sleepyhead
+					12 | 13 => if rng.gen_ratio(1, 2) { 4 } else { 2 }
                     _ => rng.sample(Uniform::new_inclusive(0, 6)),
                 };
 
+
                 level.map = rng.sample(Uniform::new_inclusive(0, 4));
+				if level.map == 4 {
+				level.behavior = rng.sample(Uniform::new_inclusive(0, 1));
+}
+	
+                 if level.map == 1 {
+                 level.behavior = match level.map {
+	             1 => if rng.gen_ratio(1, 4) { 0 } else if rng.gen_ratio(1, 2) { 1 } else { 4 }
+				 4 => if rng.gen_ratio(1, 2) { 0 } else { 1 } //Elevator Behavior Softlock Fix
+				 3 => if rng.gen_ratio(1, 2) { 0 } else { 1 } //Tennis Stands Behavior Softlock Fix
+	             _ => rng.sample(Uniform::new_inclusive(4, 5)),
+                };
+	}
```

Comment:
We should try and combine these settings of level.behavior, because each one ignores any constraints of the previous one.

i.e., after moving the generation of map above the generation of behavior:
```rust
level.behavior = match level.level_type {
    5 | 9 | 10 | 11 | 20 => if let 3 | 4 = level.map {
        1 // this is the only behavior possible with this combination of level_type and map
    } else if rng.gen_ratio(1, 2) { 1 } else { 4 },
    12 | 13 => <...>, // TODO: we need to prevent level_type from being 12 or 13 and map from being 3 or 4 at the same time.
    14 | 15 | 16 | 17 | 18 | 19 => 0, // TODO: are there any maps/level_types that cannot accept a behavior of 0?
    _ if level.map == 1 => <...,>,
    _ => if let 3 | 4 = level.map { <...> } else { <...> }
}
```

Reply:
Yeah, it was a real mess when I tried to edit and copy over the existing syntax. Also every stage can use behavior type "0"

Reply:
okay, but there are some cases farther down where you force the behavior to be 1, should those have instead been 0 or 1?

Reply:
That's for the "Find the odd mii out" objectives. They require the mii behavior to be "1"